### DISCUSSION TITLE
They don't think you can do it

### DISCUSSION THREAD

**Discussion by mindplay-dk** [rttist/rttist#15] [General]
He he

https://github.com/microsoft/TypeScript/issues/3628#issuecomment-1634916613

Challenge accepted, right? 😆✊

**Comment by Hookyns** [comment]
I don't think that's what he meant. It's just out of scope of TypeScript itself to generate metadata for reflection. So there is a room for the proper tool and it's partly positive; I'm not just loosing time for something they might do too. 🙂

**Reply by mindplay-dk** [reply]
That's a great point of view! 😊

I do hope they will eventually open up the API and officially support extensions though. I'm hoping for a future where the compiler is open and extensible enough that we won't need all these alternative compilers - where the official compiler is attractive and performant enough that people choose it over alternative compilers.
